bin/src/main.rs

```diff
@@ -71,25 +70,30 @@ async fn main() {
     let mut _agent_state = None;
     let mut offset_state = None;
     let mut initial_offsets = None;
-    if !matches!(config.log.lookback, Lookback::None) {
-        if let Some(path) = config.log.db_path {
-            match AgentState::new(path) {
-                Ok(agent_state) => {
-                    let _offset_state = agent_state.get_offset_state();
-                    let offsets = _offset_state.offsets();
-                    _agent_state = Some(agent_state);
-                    offset_state = Some(_offset_state);
-                    match offsets {
-                        Ok(os) => {
-                            initial_offsets =
-                                Some(os.into_iter().map(|fo| (fo.key, fo.offset)).collect());
+    match config.log.db_path {
```

Comment:
Would it be cleared to change this to an if let given that one of the match branches is {} ?

Reply:
+1